### PR TITLE
Improve logging for PLC read and write operations

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

**Review by olbjo — CHANGES_REQUESTED** [review]
I have some ideas for improvement. Lets discuss if these are worth implementing.

File: src/device/pf_cmrdr.c [hunk 1]
```diff
@@ -134,36 +134,106 @@ int pf_cmrdr_rm_read_ind (
          (p_read_request->index >= PF_IDX_SUB_DIAGNOSIS_CH) &&
          (p_read_request->index <= PF_IDX_SUB_DIAG_MAINT_DEM_ALL))
       {
-         index_range_text = "subslot";
+         index_range_text = "subslot diagnosis";
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmrdr.c]: Done

File: src/device/pf_cmrdr.c [hunk 2]
```diff
@@ -134,36 +134,106 @@ int pf_cmrdr_rm_read_ind (
...
+      default:
+         break;
       }
 
       if (index_range_text != NULL)
       {
          LOG_INFO (
             PNET_LOG,
-            "CMRDR(%d): PLC is reading %s diagnosis. Slot %u subslot 0x%04X "
+            "CMRDR(%d): PLC is reading %s. Slot %u subslot 0x%04X "
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmrdr.c]: Done

File: src/device/pf_cmrdr.c [hunk 3]
```diff
@@ -134,36 +134,106 @@ int pf_cmrdr_rm_read_ind (
...
+      case PF_IDX_DEV_CONN_MON_TRIGGER:
+         index_range_text = "nothing, but is restarting the timeout period";
+         break;
+
+      case PF_IDX_SUB_PDINTF_REAL:
+         index_range_text = "subslot PD interface real";
+         break;
+
+      case PF_IDX_SUB_PDINTF_ADJUST:
+         index_range_text = "subslot PD interface adjust";
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmrdr.c]: Done

**Comment by pyhys** [comment]
Rebased on top of master

**Comment by pyhys** [comment]
Rebased again

File: src/device/pf_cmrdr.c [hunk 4]
```diff
@@ -931,11 +891,17 @@ int pf_cmrdr_rm_read_ind (
          }
          break;
       default:
+#if LOG_INFO_ENABLED(PNET_LOG)
          LOG_INFO (
             PNET_LOG,
-            "cmrdr(%d): No support for reading index 0x%x\n",
+            "CMRSR(%d): No support for reading slot %u subslot 0x%04X index "
```
Comment (pyhys) [inline src/device/pf_cmrdr.c:897]: Could not read slot %u subslot ...

File: src/device/pf_pdport.c [hunk 5]
```diff
@@ -541,8 +540,25 @@ int pf_pdport_read_ind (
...
+            "for slot %u, subslot 0x%04X\n",
+            __LINE__,
+            index,
+            slot,
+            subslot);
+      }
       break;
+
    case PF_IDX_DEV_PDEXP_DATA:
+      LOG_INFO (
```
Comment (olbjo) [inline src/device/pf_pdport.c:557]: Can be removed if a common error message is added.

**Review by olbjo — CHANGES_REQUESTED** [review]
Great!
Still some comments on the log entries.

File: src/device/pf_cmrdr.c (hunk 4 above)
Comment (pyhys) [reply src/device/pf_cmrdr.c:897]: Done

File: src/device/pf_cmrdr.c [hunk 6]
```diff
@@ -25,7 +25,7 @@
  * @brief Implements the Context Management Read Record Responder protocol
  * machine device (CMRDR)
  *
- * Contains a single function \a pf_cmrdr_rm_read_ind(),
+ * Contains mainly a function \a pf_cmrdr_rm_read_ind(),
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmrdr.c:28]: Done

File: src/device/pf_cmrdr.c [hunk 7]
```diff
@@ -59,7 +59,33 @@ int pf_cmrdr_rm_read_ind (
...
 
 #if LOG_INFO_ENABLED(PNET_LOG)
-   char * index_range_text = NULL;
+   if (
+      (p_read_request->slot_number == PNET_SLOT_DAP_IDENT) &&
+      (pf_port_subslot_is_dap_port_id (p_read_request->subslot_number) == true))
+   {
+      LOG_INFO (
+         PNET_LOG,
+         "CMRDR(%d): PLC is reading slot %u subslot 0x%04X (local port %d) "
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmrdr.c:68]: Done

File: src/device/pf_cmwrr.c [hunk 8]
```diff
@@ -155,6 +155,37 @@ static int pf_cmwrr_write (
...
 
+#if LOG_INFO_ENABLED(PNET_LOG)
+   if (
+      (p_write_request->slot_number == PNET_SLOT_DAP_IDENT) &&
+      (pf_port_subslot_is_dap_port_id (p_write_request->subslot_number) ==
+       true))
+   {
+      LOG_INFO (
+         PNET_LOG,
+         "CMWRR(%d): PLC is writing slot %u subslot 0x%04X (local port %u) "
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmwrr.c:166]: Done

File: src/device/pf_pdport.c [hunk 9]
```diff
@@ -461,13 +421,26 @@ int pf_pdport_read_ind (
             p_pos);
          ret = 0;
       }
+      else
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_pdport.c:424]: Done

File: src/device/pf_pdport.c [hunk 10]
```diff
@@ -482,13 +455,25 @@ int pf_pdport_read_ind (
          }
          ret = 0;
       }
+      else
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_pdport.c:458]: Done

File: src/device/pf_pdport.c [hunk 11]
```diff
@@ -522,8 +508,21 @@ int pf_pdport_read_ind (
             ret = 0;
          }
       }
+      if (ret != 0)
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_pdport.c:511]: Done

File: src/device/pf_pdport.c [hunk 12]
```diff
@@ -541,8 +540,25 @@ int pf_pdport_read_ind (
             p_pos);
          ret = 0;
       }
+      else
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_pdport.c:543]: Done

File: src/device/pf_pdport.c (hunk 5 above)
Comment (pyhys) [reply src/device/pf_pdport.c:557]: Done

File: src/device/pf_cmrdr.c [hunk 13]
```diff
@@ -931,11 +891,17 @@ int pf_cmrdr_rm_read_ind (
          }
          break;
       default:
+#if LOG_INFO_ENABLED(PNET_LOG)
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmrdr.c:894]: Done

[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmrdr.c]: Done

File: src/device/pf_cmrdr.c [hunk 14]
```diff
@@ -931,11 +891,17 @@ int pf_cmrdr_rm_read_ind (
...
+            "CMRSR(%d): No support for reading slot %u subslot 0x%04X index "
+            "0x%04X \"%s\"\n",
             __LINE__,
-            p_read_request->index);
+            p_read_request->slot_number,
+            p_read_request->subslot_number,
+            p_read_request->index,
+            pf_index_to_logstring (p_read_request->index));
+#endif
          ret = -1;
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmrdr.c:899]: Done

**Review by olbjo — APPROVED** [review]

**Comment by pyhys** [comment]
Rebased

File: src/device/pf_cmrdr.c [hunk 15]
```diff
@@ -59,7 +59,33 @@ int pf_cmrdr_rm_read_ind (
    bool new_flag = false;
 
 #if LOG_INFO_ENABLED(PNET_LOG)
```
Comment (hefloryd) [inline src/device/pf_cmrdr.c:61]: The LOG_INFO_ENABLED check could be removed now?

File: src/device/pf_cmwrr.c [hunk 16]
```diff
@@ -155,6 +155,37 @@ static int pf_cmwrr_write (
 {
    int ret = -1;
 
+#if LOG_INFO_ENABLED(PNET_LOG)
```
Comment (hefloryd) [inline src/device/pf_cmwrr.c:158]: See above.

**Pull request merged**